BEST_PRACTICES.md

```diff
@@ -166,8 +160,171 @@ test:
+              "zip": {"type": "string", "selectable": true},
+            }
+          }
+        }
+      }
+    },
+    "orders": {
+      "type": "object",
+      "properties": {
+        "id": {"type": "integer"},
+        "user_id": {"type": "integer", "selectable": true},
+        "amount": {"type": "number", "selectable": true},
+        "credit_card_number": {"type": "string", "selectable": true},
+      }
+    }
+  }
+}
+```
 
+### Property Selection
+
+A tap that supports property selection should accept an optional
+`--properties PROPERTIES` option. `PROPERTIES` should point to a file
+containing the desired schemas.
+
+The format of PROPERTIES file is similar to the output from discovery.
+The top level is an object, with a single key called "streams", that
+points to a map where each key is a stream name and each value is the
+desired schema for that stream.
+
+The Tap should attempt to sync every stream that is listed in the
+PROPERTIES file. For each of those streams, it SHOULD include all the
+properties that are present for that stream in the desired schema. If the
+desired schema contains a property that does not exist in the data source,
+a Tap MAY fail and exit with a non-zero status or it MAY omit the
+requested field from the output. The Tap MAY include additional properties
+that are not included in the desired schema, if those properties are
+always provided by the data source. The tap MUST NOT include in its output
+any streams that are not present in desired schemas file.
+
+Note that the Tap should not validate the data against the desired schema.
+The purpose of the desired schema is _only_ to communicate which streams
+and properties are desired.
+
+For example, suppose we wanted to sync only the users table, and only the
+last name and state of each user. Our desired schemas file would look like
+this:
+
+```javascript
+{
+  "streams": {
+    "users": {
+      "type": "object",
+      "properties": {
+        "id": {"type": "integer"},
+        "name": {
+          "type": "object",
+          "properties": {
+            "last_name": {"type": "string"}
+          },
+        },
+        "addresses": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "state": {"type": "string"}
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
 
-
-
-
+Since the Tap will ignore the type information on the desired schema, we
+can prune out the "type" fields. The following is sufficient for a Tap to
+know that it should sync the "users.last_name" and "addresses.state"
+properties of the "users" stream:
+
+```javascript
+{
+  "streams": {
+    "users": {
+      "properties": {
+        "name": {
+          "properties": {
+            "last_name": {}
+          },
+        },
+        "addresses": {
+          "items": {
+            "properties": {
+              "state": {}
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
```

Comment:
@iterati this does not take the place of the schema that the tap already uses, and it doesn't change the protocol between the tap and target. This is just for the purpose of letting the user communicate which fields they want. Every tap must still emit a full schema exactly as they do now.